models/mpas_atm/model_mod.f90

```diff
@@ -5017,11 +5100,19 @@ subroutine convert_vert_distrib_state(state_handle, ens_size, location, quantity
         call compute_scalar_with_barycentric (state_handle, ens_size, location(1), 3, ivars, values, istatus)
 
         ! Convert theta, rho, qv into pressure
-        call compute_full_pressure(ens_size, values(1, :), values(2, :), values(3, :), fullp(:), tk(:), istatus(:))
-        if (debug > 9 .and. do_output()) then
-          write(string2,'("zout_full_pressure, theta, rho, qv:",3F10.2,F18.8)') fullp, values(1:3,1)
-          call error_handler(E_MSG, 'convert_vert_distrib_state',string2,source, revision, revdate)
-        endif
+        call compute_full_pressure(ens_size, values(1,:), values(2,:), values(3,:), fullp, tk, istatus)
+        if( all(istatus /= 0) ) then
+           if (debug > 1 .and. do_output()) then
+           write(string2,'("Failed in compute_full_pressure")')
+           call error_handler(E_MSG, 'convert_vert_distrib_state',string2,source, revision, revdate)
+           endif
+        else
+           if (debug > 9 .and. do_output()) then
+           write(string2,'("zout [Pa],theta,rho,qv:",3F10.2,F12.8)') fullp(1), values(1:3,1)
+           call error_handler(E_MSG, 'convert_vert_distrib_state',string2,source, revision, revdate)
```

Comment:
See following comment about 'zout' ... minimally, string2 should contain something about it only printing the infor for ensemble member 1.

Reply:
see my earlier comment about the dangers of printing zout vs just zout(1).  i think zout(1) is sufficient for most purposes but putting something in the string about member 1 is a good idea.